pkg/basicstation/cups/messages.go

```diff
@@ -123,8 +123,10 @@ func (r UpdateInfoResponse) MarshalBinary() ([]byte, error) {
 	} else {
 		return nil, errFieldLength.WithAttributes("field", "tcCred", "length", credLen, "maximum", math.MaxUint16)
 	}
-	if sigLen := len(r.Signature); sigLen <= math.MaxUint16 {
-		binary.LittleEndian.PutUint16(lenBytes, uint16(sigLen))
+	// NOTE: Unlikely that the signature length exceeds 32bits (or even 16 for that matter) so this is safe to do.
+	if sigLen := uint64(len(r.Signature)); sigLen <= math.MaxUint32 {
```

Comment:
What is `uint64()` for here?

Reply:
For `x86` without this upcast the condition `sigLen <= math.MaxUint32` will always be true.